pkg/adsc/adsc.go

```diff
@@ -539,25 +614,55 @@ func (a *ADSC) handleRecv() {
 
 		// TODO: add hook to inject nacks
 
-		a.mutex.Lock()
-		if len(gvk) == 3 {
-			gt := resource.GroupVersionKind{Group: gvk[0], Version: gvk[1], Kind: gvk[2]}
-			a.sync[gt.String()] = time.Now()
-		}
-		a.Received[msg.TypeUrl] = msg
-		a.ack(msg)
-		a.mutex.Unlock()
+		a.onReceive(msg)
 
 		if len(listeners) > 0 {
+			if a.LocalCacheDir != "" {
+				strResponse, err := json.MarshalIndent(listeners, "  ", "  ")
+				if err == nil {
+					err = ioutil.WriteFile(a.LocalCacheDir+"_lds.json", strResponse, 0644)
```

Comment:
If `LocalCacheDir` is not a directory but just a base name perhaps the name should be changed?

Reply:
Will do that. Do you think it should be a dir ?